src/NuGet.Clients/NuGet.PackageManagement.UI/Models/PackageItemListViewModel.cs

```diff
@@ -640,21 +640,34 @@ private void TriggerStatusLoader()
 
         private async System.Threading.Tasks.Task ReloadPackageVersionsAsync()
         {
-            var result = await _backgroundLatestVersionLoader.Value;
-
-            await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
Guess you determined we're always on the UI thread anyway?

Reply:
let me go review codepath and confirm again.

Reply:
The threading rules say don't make assumptions :)
If already on the UI thread, the switch call it extremely cheap.

Reply:
i put those calls back in. this is not the focus of the bug that i'm trying to fix, so i shouldn't get distracted by it.

you only need to be on the ui thread when talking to wpf objects, talking to datamodels which are databound are fine to be done on background thread. so these can likely be cleanup when it makes sense.